barretenberg/cpp/src/barretenberg/benchmark/ultra_bench/mock_circuits.hpp

```diff
@@ -23,7 +23,7 @@ template <typename Builder> void generate_basic_arithmetic_circuit(Builder& buil
     stdlib::field_t a(stdlib::witness_t(&builder, fr::random_element()));
     stdlib::field_t b(stdlib::witness_t(&builder, fr::random_element()));
     stdlib::field_t c(&builder);
-    size_t passes = (1UL << log2_num_gates) / 4 - 4;
+    size_t passes = (1UL << log2_num_gates) / 4 - 8;
```

Comment:
why?

Reply:
I will add a comment, the function fills a circuit up ot the desired numbers of gates for the benchmark, ensuring finalisation and the add gates to ensure non zero polys method wont cause it to go to the next power of 2 (or at least that was my deduction). Because we enabled poseidon add gates to ensure non zero polys will add more gates so i need to lower a bit the number of passes